Proteomics/ProteolyticDigestion/DigestionMotif.cs

```diff
@@ -120,39 +120,47 @@ public bool Fits(string sequence, int location)
             // check for inducing cleavage
             for (m = 0; m < InducingCleavage.Length && fits; m++) // handle patterns
             {
-                currentResidue = sequence[location + m];
-                if (!MotifMatches(InducingCleavage[m], currentResidue))
+                if (location + m >= sequence.Length)
                 {
                     fits = false;
                 }
+                else
+                {
+                    currentResidue = sequence[location + m];
+                    if (!MotifMatches(InducingCleavage[m], currentResidue))
+                    {
+                        fits = false;
+                    }
+                }
             }
 
+            bool prevents = false;
             // check for preventing cleavage
             if (fits && PreventingCleavage != null)
             {
-                bool match = true;
-                for (int n = 0; n < PreventingCleavage.Length && match; n++)
+                prevents = true;
+                for (int n = 0; n < PreventingCleavage.Length && prevents; n++)
                 {
-                    if (location + m + n > sequence.Length || location - PreventingCleavage.Length + 1 + n < 0)
+                    if (location + m + n >= sequence.Length || location - PreventingCleavage.Length + 1 + n < 0)
                     {
-                        match = false;
+                        prevents = false;
                     }
                     else
                     {
-                        currentResidue = CutIndex != 0 ? sequence[location + m + n] : sequence[location - PreventingCleavage.Length + 1 + n];                      
+                        currentResidue = CutIndex != 0 ? sequence[location + m + n] : sequence[location - PreventingCleavage.Length + 1 + n];
                         if (!PreventingCleavage[n].Equals(currentResidue))
```

Comment:
`if (!PreventingCleavage[n].Equals(currentResidue))`
change to
`if(MotifMatches(PreventingCleavage[n], currentResidue))`

Reply:
add a unit test for this (wildcard in the preventing cleavage motif)